File: 프로그래머스/unrated/120852. 소인수분해/소인수분해.py
def divide(n):
    cnt = 0
    for i in range(1,n+1):
        if n%i==0:
            cnt += 1
    return cnt

def solution(n):
    answer = []
    for i in range(1,n+1):
        if divide(i)==2 and n%i==0:
            answer.append(i)
    
    return answer

File: 프로그래머스/unrated/120852. 소인수분해/소인수분해.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public boolean isSoin(int n){
        int cnt = 0;
        
        for(int i=1;i<=n;i++){
            if(n%i == 0)
                cnt++;
        }
        if(cnt == 2) return true;
        else return false;
    }
    
    public int[] solution(int n) {
        List<Integer> answer = new ArrayList<Integer>();
        
        for(int i=2;i<=n;i++){
            if(isSoin(i) && n%i == 0){
                answer.add(i);
            }
        }
        
        int[] result = answer.stream()
            .mapToInt(Integer::intValue)
                .toArray();
        
        return result;
    }
}